Lib/pydoc.py

```diff
@@ -1617,13 +1617,10 @@ def pipepager(text, cmd):
 def tempfilepager(text, cmd):
     """Page through text by invoking a program on a temporary file."""
     import tempfile
-    filename = tempfile.mktemp()
-    with open(filename, 'w', errors='backslashreplace') as file:
+    with tempfile.NamedTemporaryFile('w', errors='backslashreplace') as file:
         file.write(text)
-    try:
-        os.system(cmd + ' "' + filename + '"')
-    finally:
-        os.unlink(filename)
+        file.flush()
+        os.system(cmd + ' "' + file.name + '"')
```

Comment:
I don't think we can (or should) rely on this working while we still have the file open, which unfortunately means that NamedTemporaryFile as it stands isn't a great alternative. There are some open proposals to improve it though, so perhaps soon.

Reply:
Yes, `NamedTemporaryFile` can't be used like this so long as `delete=True` is implemented only with delete-on-close semantics, which opens the file with delete access. Opening it again thus requires sharing delete access, which is rarely shared. The default pager "more.com" does not share delete access.

Reply:
We can use `TemporaryDirectory`.

```
with tempfile.TemporaryDirectory() as tempdir:
    filename = os.path.join(tempdir, 'pydoc.out')
    ...
```